web/source/osk/oskManager.ts

```diff
@@ -877,6 +984,39 @@ namespace com.keyman.osk {
       }
     }
 
+    public refreshLayout(): void {
```

Comment:
Big, important piece of the new paradigm.

Existing OSK methods:
- `_Load`:  whenever a new keyboard becomes active, this "loads" the correct configuration for the keyboard.
    - In the past, also performed some of the OSK's layout operations.
- `_Show`:  responsible for displaying the keyboard.
    - In the past, also performed some of the OSK's layout operations.

While not 100% complete in this PR, one of the main goals here is to splinter those 'layout operation' parts into _this_ method:  `refreshLayout`.  It'll be much more maintainable when all layout aspects are centrally-defined and modularized.

This is designed largely after a common UI lifecycle paradigm.  As expressed in iOS / Swift:
- `will`/`did` `Load`
- `will`/`did` `Appear` (show)
- `layoutIfNeeded`

Even now, KMW actually requires _rebuilding the `VisualKeyboard` from scratch_ whenever the keyboard is rotated.  That's... not exactly efficient.  This'll get us one step closer to being able to reuse the old layout rather than tossing it outright.  (Main remaining issue:  `_Load` (for touch) and cookies (for desktop) are responsible for setting the `width` and `height` properties.)